### ISSUE TITLE
task.robbing cannot handle abandoned structures

### ISSUE THREAD

**Issue by karlthepagan** [ScreepsOCS/screeps.behaviour-action-pattern#657]
https://screeps.com/a/#!/history/E48N82?t=17167802

If that data is saved you'll see a pair or robbers circling around the terrain at the center of the room. These were trying to walk onto the flag that was their travel target.

There are several links, storage, and an extension which should be priority robbing targets.

This was a room abandoned and I'm trying to recover the energy.

**Comment by pandabear41** [comment]
@karlthepagan Was there a rampart under the structure?

**Comment by karlthepagan** [comment]
there are no ramparts in my abandoned room

**Comment by Dasku** [comment]
I'm testing this now with the proposed master branch and its working fine. Robbers emptied all the extensions and now are robbing the storage as expected.  Close for now?

**Comment by karlthepagan** [comment]
:+1:

**Comment by karlthepagan** [comment]
Must be a problem with canary. Some time after this my robbers sat and did nothing

https://screeps.com/a/#!/history/E71S7?t=17396239

**Comment by karlthepagan** [comment]
Yeah I think that's the issue. I use robbing a lot to try to pull from abandoned rooms.
